WordPress/src/main/java/org/wordpress/android/ui/prefs/categories/CategoriesListFragment.kt

```diff
@@ -45,10 +54,78 @@ class CategoriesListFragment : Fragment(R.layout.site_settings_categories_list_f
 
     private fun SiteSettingsCategoriesListFragmentBinding.setupObservers() {
         viewModel.uiState.observe(viewLifecycleOwner, {
-            // todo handle the ui states
+            when (it) {
+                is Content -> showList(it.list)
+                is Error -> showError(it)
+                is Loading -> showLoading()
+            }
         })
     }
 
+    private fun SiteSettingsCategoriesListFragmentBinding.showLoading() {
+        progressBar.updateVisibility(true)
+
+        fabButton.updateVisibility(false)
+        categoriesRecyclerView.updateVisibility(false)
+        actionableEmptyView.updateVisibility(false)
+    }
+
+    private fun SiteSettingsCategoriesListFragmentBinding.showError(error: Error) {
+        updateErrorContent(error)
+        actionableEmptyView.updateVisibility(true)
+
+        fabButton.updateVisibility(false)
+        progressBar.updateVisibility(false)
+        categoriesRecyclerView.updateVisibility(false)
+    }
```

Comment:
@ashiagr Great suggestion 🙌 . I had seen this logic but thought it would be an overkill. I see the point now, by updating the `UiSstate` to hold that logic, the `Fragment` looks better.  Thank you for this suggestion. 👍 
Done in https://github.com/wordpress-mobile/WordPress-Android/pull/15580/commits/e620061108e8a176167ec2046883445641410d0e